vault/diagnose/tls_verification.go

```diff
@@ -21,17 +20,32 @@ const maxVersionError = "'tls_max_version' value %q not supported, please specif
 
 // ListenerChecks diagnoses warnings and the first encountered error for the listener
 // configuration stanzas.
-func ListenerChecks(ctx context.Context, listeners []listenerutil.Listener) ([]string, []error) {
+func ListenerChecks(ctx context.Context, listeners []*configutil.Listener) ([]string, []error) {
+	testName := "check-listener-tls"
+	ctx, span := StartSpan(ctx, testName)
+	defer span.End()
 
 	// These aggregated warnings and errors are returned purely for testing purposes.
 	// The errors and warnings will report in this function itself.
 	var listenerWarnings []string
 	var listenerErrors []error
 
-	for _, listener := range listeners {
-		l := listener.Config
+	for _, l := range listeners {
 		listenerID := l.Address
 
+		if l.TLSDisable {
+			Warn(ctx, fmt.Sprintf("listener at address: %s has error %s. ", listenerID, "TLS is disabled in a Listener config stanza"))
+			continue
+		}
+		if l.TLSDisableClientCerts {
+			Warn(ctx, fmt.Sprintf("listener at address: %s has error %s. ", listenerID, "TLS for a listener is turned on without requiring client certs"))
+
+		}
+		status, warning := TLSMutualExclusionCertCheck(l)
+		if status != 0 {
```

Comment:
minor, but, would it make sense to not use the negation? like : " if status == 1"
Since we are printing a warning, and we have three status values. This would avoid confusion.

Reply:
Definitely!